packages/create-llama/helpers/env-variables.ts

```diff
@@ -0,0 +1,154 @@
+import fs from "fs/promises";
+import path from "path";
+import {
+  FileSourceConfig,
+  TemplateDataSource,
+  TemplateFramework,
+  TemplateVectorDB,
+  WebSourceConfig,
+} from "./types";
+
+const renderEnvVar = (
+  name: string,
+  description?: string,
+  value?: string,
+): string => {
+  let content = "";
+  // Add the description if it exists.
+  if (description) {
+    content += `# ${description.replaceAll("\n", "\n# ")}\n`;
+  }
+  // Set the value if it exists, otherwise comment it out.
+  content += value && value != "" ? `${name}=${value}\n\n` : `# ${name}=\n\n`;
+  return content;
+};
+
+export const createEnvLocalFile = async (
+  root: string,
+  opts?: {
+    openAiKey?: string;
+    llamaCloudKey?: string;
+    vectorDb?: TemplateVectorDB;
+    model?: string;
+    embeddingModel?: string;
+    framework?: TemplateFramework;
+    dataSource?: TemplateDataSource;
+  },
+) => {
+  const envFileName = ".env";
+  let content = "";
+
+  const model = opts?.model || "gpt-3.5-turbo";
+  content += `MODEL=${model}\n`;
+  if (opts?.framework === "nextjs") {
+    content += `NEXT_PUBLIC_MODEL=${model}\n`;
+  }
+  console.log("\nUsing OpenAI model: ", model, "\n");
+
+  if (opts?.openAiKey) {
+    content += `OPENAI_API_KEY=${opts?.openAiKey}\n`;
+  }
+
+  if (opts?.framework === "fastapi") {
+    //  Default environment variables for FastAPI
+    const defaultEnvs = [
```

Comment:
great idea with this defaultEnvs - how about we put all variables in there first and then render the string?
from what I see, only conditional rendering is missing, we could do this in two ways:

1. add a `render` flag to each entry, e.g.:
```js
[
{
        render: opts?.framework === "nextjs",
        name: "NEXT_PUBLIC_MODEL",
        value: opts?.model ?? "gpt-3.5-turbo",
}
...
 ```
 or decide to add the entry in the array literal:
 ```js
[
   ...opts?.framework === "nextjs" ? [{
        name: "NEXT_PUBLIC_MODEL",
        value: opts?.model ?? "gpt-3.5-turbo",
    }] : []
...
 ```
 
I guess we could then separate building the structure and rendering it. Also we could make two defaultEnvs: one for python and one for TS (and therefore separate each environment then nicely)

Reply:
Thanks for the suggestion, it's great to apply to all our default env variables.